File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-java-Mothyque/backend/Common/src/main/java/org/example/dto/Response.java
package org.example.dto;

import java.io.Serializable;

public class Response implements Serializable
{
    private final ResponseType type;
    private final Object data;

    public Response(ResponseType type, Object data)
    {
        this.type = type;
        this.data = data;
    }

    public ResponseType getType()
    {
        return type;
    }

    public Object getData()
    {
        return data;
    }
}

File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-csharp-Mothyque/ProiectMPP/ProiectMPP.Common/Dto/Response.cs
using System;

namespace ProiectMPP.Common.dto
{
    [Serializable]
    public class Response
    {
        private readonly ResponseType type;
        private readonly object data;

        public Response(ResponseType type, object data)
        {
            this.type = type;
            this.data = data;
        }

        public ResponseType Type
        {
            get { return type; }
        }

        public object Data
        {
            get { return data; }
        }
    }
}